desk/lib/channel-utils.hoon

```diff
@@ -676,4 +1055,159 @@
       ;br;
     ==
   --
+::
+++  v-channels-7-to-8
+  |=  vc=v-channels:v7:old:c
+  ^-  v-channels:c
+  %-  ~(run by vc)
+  |=  v=v-channel:v7:old:c
+  ^-  v-channel:c
+  =/  [log=log:v-channel:c mod=(map id-post:c time)]
+    (log-7-to-8 log.v)
+  =/  [count=@ud =v-posts:c]
+    (v-posts-7-to-8 posts.v mod)
+  =-  -(|1.- [count |1.-<])
+  %=  v
+    posts    v-posts
+    log      log
+    future   (future-8 log)
+  ==
+++  v-posts-7-to-8
+  |=  [vp=v-posts:v7:old:c mod=(map id-post:c time)]
+  ^-  [@ud v-posts:c]
+  =|  posts=v-posts:c
+  %+  roll  (tap:on-v-posts:v7:old:c vp)
+  |=  [[=id-post:c post=(unit v-post:v7:old:c)] count=@ud =_posts]
+  ^+  [count posts]
+  ::  for each post traversed, even if it was deleted,
+  ::  .count increases to generate correct post sequence
+  ::  number
+  ::
+  =.  count  +(count)
+  :-  count
+  ?~  post  posts
+  ::  insert seq and mod-at into seal
+  ::
+  =/  new-post=v-post:c
+    =/  new-seal=v-seal:c
+      =+  mod-at=(~(got by mod) id-post)
```

Comment:
this crashes currently because we don't capture logs in `%channels` agent only `%channels-server`, so when we try to get the time here it just bails and then fails to load. I think for now we can just default to either the post id or the latest reply id. maybe we want to modify `ca-u-channels` to capture and keep the logs, because we'll likely need it for changes endpoints later on.

Reply:
As we talked, we default to post id in the absence of corresponding log event.